hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java

```diff
@@ -216,19 +216,19 @@ public boolean equals(Object obj) {
     /**
      * Array of target FileSystem URIs.
      */
-    private final URI[] targetFileSystemURIs;
+    private final String[] targetFileSystemPaths;
 
-    MountPoint(Path srcPath, URI[] targetFs) {
+    MountPoint(Path srcPath, String[] targetFs) {
       mountedOnPath = srcPath;
-      targetFileSystemURIs = targetFs;
+      targetFileSystemPaths = targetFs;
     }
 
     public Path getMountedOnPath() {
       return mountedOnPath;
     }
 
-    public URI[] getTargetFileSystemURIs() {
-      return targetFileSystemURIs;
+    public String[] getTargetFileSystemPaths() {
```

Comment:
This is a public method, can we keep the old signature? We can transform from string to URI within the method itself.

Reply:
Added getTargetFileSystemURIs where we are returning URI[]. Also added method to return targetFileSystem Paths.